src/rest/type.pug

```diff
@@ -42,15 +45,18 @@ mixin typeNameString(type, skipRequired)
         if type.default
             span.help(title='Default Value')= `=${type.default}`
     else if union
-        | (
         for nested, i in union
             if i > 0
                 = ' ∪ '
-            if rest.types[nested]
-                a(href=`#${nested}`)= nested
+            -
+                var nestedType = nested;
+                if (nestedType.slice(-2) === '[]') {
+                    nestedType = nested.slice(0,-2);
+                }
+            if rest.types[nestedType]
+                a(href=`#${nestedType}`)= nested
```

Comment:
This is sort of a duplicate of above code. Seems like a simple recursive sort of thing would be useful here.

```python
def print_type(name):
  if is_union(name):
    return ' | '.join(print_type(t) for t in name.split(' | '))
  
  # everything else...
      
```